### PR TITLE
[Resources] move resources definition to php

### PR DESCRIPTION
**Pull request by SirDomin**
| Q               | A
| --------------- | -----
| Branch?         | master
| Bug fix?        | no
| New feature?    | no
| Related tickets | 

[Resources] move resources definition to php

### CODE REVIEW

File: src/Resources/config/app/config.yml [hunk 1]
```diff
@@ -7,35 +7,6 @@ sylius_refund:
         allowed_files:
             - '%env(default:default_logo_file:resolve:SYLIUS_REFUND_LOGO_FILE)%'
 
-sylius_resource:
```
Comment (GSadee) [inline src/Resources/config/app/config.yml:10]: As a way of customizing resources definitions will change, you should probably open this PR against 2.0 branch with a proper note to UPGRADE-2.0 file

File: tests/DependencyInjection/SyliusRefundConfigurationTest.php [hunk 2]
```diff
@@ -34,6 +53,7 @@ public function it_does_not_define_any_allowed_files_by_default(): void
     /** @test */
     public function it_allows_to_define_allowed_files(): void
     {
+
```
Comment (coldic3) [inline tests/DependencyInjection/SyliusRefundConfigurationTest.php]:
```suggestion
```

**Review by coldic3 — APPROVED** [review]

File: UPGRADE-2.0.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+### UPGRADE FROM 1.X TO 2.0
+
+1. Resources configuration have been changed from yaml to php.
```
Comment (GSadee) [inline UPGRADE-2.0.md]: You should describe here the difference in configuration for the end application

File: src/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -48,4 +68,129 @@ private function addPdfGeneratorSection(ArrayNodeDefinition $node): void
             ->end()
         ;
     }
+
+    private function addResourceSection(ArrayNodeDefinition $node): void
```
Comment (GSadee) [inline src/DependencyInjection/Configuration.php]:
```suggestion
    private function addResourcesSection(ArrayNodeDefinition $node): void
```

**Review by GSadee — APPROVED** [review]

**Review by lchrusciel — APPROVED** [review]

File: tests/DependencyInjection/SyliusRefundConfigurationTest.php [hunk 5]
```diff
@@ -61,8 +80,81 @@ public function it_allows_to_disable_pdf_generator(): void
...
         return new Configuration();
     }
+
+    private function getExpectedResourceConfiguration(): array
+    {
+        return [
+            'credit_memo' => [
+                'classes' => [
+                    'model' => CreditMemo::class,
+                    'interface' => CreditMemoInterface::class,
```
Comment (Roshyo) [inline tests/DependencyInjection/SyliusRefundConfigurationTest.php:104]: Shouldn't the interface be auto-discovered by Sylius ?

**Pull request merged**